Bump istio installation

aerosouund opened the pull request:
**What this PR does / why we need it**:

Bump Istio to the latest stable release

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1324


**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

mhenriks commented:
@aerosouund thanks for taking a crack at this! I am very keen to get this in. Do you need any help? The current version we are using is very old, and is not compatible with https://github.com/kubevirt/kubevirt/pull/13422. The istio sidecar injection code is somehow stealing the `runStrategy: Always` bit from `virt-tail`. And since the istio container is not meant to always run, the virt-luancher pod never gets to running state.

cc @EdDev

aerosouund commented:
@mhenriks 
Thanks alot, I am facing a small challenge where after installation, checking the `IstioOperator` resource says that such a resource doesn't exist.
I am suspecting its because many things changed between the version we had and the version i am introducing `1.15` >> `1.24`
I just need to check whats going on, but likely its not a big deal. I will ping you if i need any assistance!

aerosouund commented:
On inspecting the cluster post running `istioctl install -y` (the `operator init` has been deprecated) it seems it the `IstioOperator` resource is no longer used

```bash
[vagrant@node01 ~]$ sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf api-resources | grep istio
wasmplugins                                      extensions.istio.io/v1alpha1      true         WasmPlugin
destinationrules                    dr           networking.istio.io/v1            true         DestinationRule
envoyfilters                                     networking.istio.io/v1alpha3      true         EnvoyFilter
gateways                            gw           networking.istio.io/v1            true         Gateway
proxyconfigs                                     networking.istio.io/v1beta1       true         ProxyConfig
serviceentries                      se           networking.istio.io/v1            true         ServiceEntry
sidecars                                         networking.istio.io/v1            true         Sidecar
virtualservices                     vs           networking.istio.io/v1            true         VirtualService
workloadentries                     we           networking.istio.io/v1            true         WorkloadEntry
workloadgroups                      wg           networking.istio.io/v1            true         WorkloadGroup
authorizationpolicies               ap           security.istio.io/v1              true         AuthorizationPolicy
peerauthentications                 pa           security.istio.io/v1              true         PeerAuthentication
requestauthentications              ra           security.istio.io/v1              true         RequestAuthentication
telemetries                         telemetry    telemetry.istio.io/v1             true         Telemetry
```

With that said, the yaml files we use to install the operator will no longer work (`istio-operator.cr.yaml` & `istio-operator-with-cnao.cr.yaml`)
This signifies a huge migration from what we have, previously what you would have in the cluster is

```bash
[vagrant@node01 ~]$ sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -A | grep istio
istio-operator   istio-operator-6c4fc4d784-2qqff            1/1     Running   0          2m50s
istio-system     istio-egressgateway-79c995f7cb-w4ppg       1/1     Running   0          88s
istio-system     istio-ingressgateway-775fdbc456-qq5cn      1/1     Running   0          88s
istio-system     istiod-5857496459-cnz5t                    1/1     Running   0          92s
kube-system      istio-cni-node-l99td                       1/1     Running   0          88s
```

After the upgrade, you only get

```bash
[vagrant@node01 ~]$ sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -A | grep istio
istio-system   istio-ingressgateway-5f9df778cc-bl9sw      1/1     Running   0          33m
istio-system   istiod-69d6bb74c-z6fqk                     1/1     Running   0          34m
```

So we need to know how to get the same as what we had before in the previous version using 1.24. I might need help from the network team on this

cc: @oshoval @EdDev

aerosouund commented:
Alternatively, we may not jump to 1.24. Maybe a lesser version that still behaves similar to 1.15 and has the things you want supported.
Needs a discussion for sure

oshoval commented on cluster-provision/k8s/provision.sh:
```diff
@@ -4,7 +4,7 @@
 
 PHASES_DEFAULT="linux,k8s"
-PHASES="${PHASES:-$PHASES_DEFAULT}"
+PHASES="$PHASES_DEFAULT"
```
please drop, not related it seems

aerosouund replied:
Yeah this is just for debugging, will drop once the PR is ready

brianmcarey replied:
I think we still want to be able to select what phases we provision and not just set the default here - can you add it back?

oshoval commented:
I am not familiar at all with Istio tbh, and currently on few other tasks, 
if discussion will be still needed, i will try once i can, thanks

aerosouund commented:
@oshoval 
based on what changed across the deployments three questions need answered:
- is the egressgateway still needed or used ? has it's functionality been ported to any of the other components ?
- is the istiocni now being installed separately from the operator ? is yes, how ?
- would it be better to not skip to a very high version and upgrade to something intermediate? e.g: 1.18

oshoval commented:
the first two need deeper understand that i dont have now sorry
if the 3rd help advance, i will go for that first thing first, wdyt?

aerosouund commented:
@mhenriks 
whats the minimum compatible version that will contain the features you want ?

mhenriks commented:
> @mhenriks whats the minimum compatible version that will contain the features you want ?

1.20 is the oldest version that supports 1.29 (SideCar featuregate enabled) so that would be the minimum. But obviously latest would be best

https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases

ormergi commented:
>     * is the egressgateway still needed or used ? has it's functionality been ported to any of the other components ?
> 
It seem we dont use in in e2e tests, we do use  ingress-gateway though Istio API 
https://github.com/kubevirt/kubevirt/blob/94d03645710c55dd35cf752bf6709bfb333518ad/tests/network/vmi_istio.go#L640
https://istio.io/latest/docs/setup/additional-setup/gateway/#deploying-a-gateway

>     * is the istiocni now being installed separately from the operator ? is yes, how ?
> 
From what I remember Istio operator account for deploying its CNI, we have no dedicated scripting for doing that.
Check out https://istio.io/latest/docs/setup/additional-setup/cni/#installing-the-cni-node-agent

>     * would it be better to not skip to a very high version and upgrade to something intermediate? e.g: 1.18
> 
Please note kubevirt e2e tests relays on the sidecar injection functionally.

I suggest to test this PR on kubevirt/kubevirt on sig-network lane so we can see where it fails and realize what our options.

aerosouund commented:
@mhenriks @ormergi 
Ok, based on this i will move the check for the `IstioOperator` resource (which is what's failing the tests here) and devise a way to install the CNI. Then we can run it against the sig network lane and see how it goes

brianmcarey commented:
/test all

brianmcarey commented on cluster-provision/k8s/1.29/extra-pre-pull-images:
```diff
@@ -14,6 +14,9 @@ quay.io/kubevirt/cdi-apiserver:v1.58.1
 quay.io/kubevirt/cdi-uploadproxy:v1.58.1
+docker.io/istio/install-cni:1.24.1
+docker.io/istio/pilot:1.24.1
+docker.io/istio/proxyv2:1.24.1
```
I need to these images copied over to our quay account so that we don't hit any docker io limits

brianmcarey replied:
The older images at the top of the file can be removed

```
quay.io/kubevirtci/install-cni:1.15.0
quay.io/kubevirtci/operator:1.15.0
quay.io/kubevirtci/pilot:1.15.0
```

ormergi replied:
Asking to understand: @brianmcarey from what I know we have container image cache on CI, is it correct? 
Is so, do we still need mirroring if we have cache?

aerosouund commented:
@brianmcarey 
Thanks for looking at this, we also need the linux base image to be updated for the newer istio to be installed. Currently i am running the linux phase in CI for the code to work
```bash
PHASES="$PHASES_DEFAULT"
```
We can merge it as it is and then i will submit a tiny PR that reverts this change so we can go back to running the k8s phase in CI only

mhenriks commented:
/retest-required

aerosouund commented:
@mhenriks 
CI is looking good now, whats next ? is the code acceptable in its current state or do you have nay remarks ?

akalenyu commented:
@mhenriks is out, but maybe @ormergi can help driver this further
BTW the PR title is a little bit misleading, since you're bumping the istio installation as a whole and not the cli tool

ormergi commented:
@aerosouund please make sure istio bump pass all kubevirt/kubevirt CI lanes, specifically sig-network lanes.
You can do that by posting a draft PR on kubevirt/kubevirt with this PR changes.

aerosouund commented:
@ormergi This PR changes the gocli and the packages installed on the provider images in the provisioning phase. My assumption is that new versions for the gocli and providers need to be built and published on quay to make this testable on kv/kv since only the `cluster-up` part gets copied to kv/kv.
If those assumptions are false let me know and any suggestions you have on how we can test this

akalenyu commented:
> @ormergi This PR changes the gocli and the packages installed on the provider images in the provisioning phase. My assumption is that new versions for the gocli and providers need to be built and published on quay to make this testable on kv/kv since only the `cluster-up` part gets copied to kv/kv. If those assumptions are false let me know and any suggestions you have on how we can test this

It's actually possible to test prior to that - https://github.com/kubevirt/kubevirtci/blob/main/KUBEVIRTCI_LOCAL_TESTING.md

mhenriks commented:
/retest-required

ormergi commented:
Hi, I managed to make istio-1.24 work on kubevirtci cluster, and verified istio tests are passing.
Including on kubevirt that is build with https://github.com/kubevirt/kubevirt/pull/13422.

While troubleshooting the issues with 1.24 on kubevirtci, I saw istio-cni is not deployed as privileged , result VM creation failures.The privileged change is introduced in 1.24.
Istio 1.23 should work as expected, I suggest using this version.

In addition, IstioOperator API has been changed, see bellow config that worked for me:
```
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istio-operator
spec:
  profile: demo
  components:
    cni:
      enabled: true
      namespace: kube-system
  values:
    global:
      jwtPolicy: first-party-jwt
    cni:
      chained: false
      cniBinDir: /opt/cni/bin
      cniConfDir: /etc/cni/multus/net.d
      privileged: true
      excludeNamespaces:
       - istio-system
       - kube-system
      logLevel: debug
    pilot:
      cni:
        enabled: true
        provider: "multus"
    sidecarInjectorWebhook:
      injectedAnnotations:
        k8s.v1.cni.cncf.io/networks: istio-cni
```
The missing configuration are:
- `spec.values.pilot.cni`
  This is a new API for triggering CNI deployment.
   The filed `provider=multus` acts like the known `chained=false` option (maybe in can be removed, need to verify).
- `sidecarInjectorWebhook`
  On istio >= 1.19, istio will append 'default/istio-cni' to the pod's  `k8s.v1.cni.cncf.io/networks` is exist. Result in VMs failures.
  This cant  work for kubevirtci env because we do not create the `istio-cni` NAD in the default namespace, but at the test namespace.
  In general it seems wrong because it means all pods in the cluster can connect to the mesh, becasuse all pods can connect to a NAD resied in default namespaces, no isolation allowed.

aerosouund commented:
@ormergi Thanks alot for your help on this

> Istio 1.23 should work as expected, I suggest using this version.

Acknowledged, thanks.

> This cant work for kubevirtci env because we do not create the istio-cni NAD in the default namespace, but at the test namespace.
In general it seems wrong because it means all pods in the cluster can connect to the mesh, becasuse all pods can connect to a NAD resied in default namespaces, no isolation allowed.

Not sure i quite follow through but in general i will test this on kubevirtCI and check if it produces the desired results

brianmcarey commented:
Hey @aerosouund , I was wondering what the status of this one is? Have you tried with istio 1.23?

brianmcarey commented:
> > on kubevirt istio tests. But i am not sure if the problem is local to my environment or there's an actual problem is in istio. I will rewrite to 1.23 shortly
> 
> I can try and help to get a provider published once the changes are in place and we can test it against the actual CI environment to see how it looks.

I published a provider for this and ran the network e2e suite against it - it looks like the istio tests are still failing against this updated version. Attaching the artifacts from the run:
[artifacts20250204.tar.gz](https://github.com/user-attachments/files/18668883/artifacts20250204.tar.gz)

The images should be available from my quay repo
`quay.io/kubevirtci/gocli:20250204-istiobump`
`quay.io/bcarey/k8s-1.31:20250204-istiobump`

ormergi commented:
> > > on kubevirt istio tests. But i am not sure if the problem is local to my environment or there's an actual problem is in istio. I will rewrite to 1.23 shortly
> > 
> > 
> > I can try and help to get a provider published once the changes are in place and we can test it against the actual CI environment to see how it looks.
> 
> I published a provider for this and ran the network e2e suite against it - it looks like the istio tests are still failing against this updated version. Attaching the artifacts from the run: [artifacts20250204.tar.gz](https://github.com/user-attachments/files/18668883/artifacts20250204.tar.gz)
> 
> The images should be available from my quay repo `quay.io/bcarey/gocli:20250204-istiobump` `quay.io/bcarey/k8s-1.31:20250204-istiobump`

It seems tests that verifies Isito & passt binding plugin are failing, tests that verifies masquerade binding pass.
Tests that specify port forward 22 explicitly in the VMI spec fails (7/8) because the VM didn't become ready, due to the following error coming form passt process in the VMI virt-launcher pod:
```
<failure type="Failure">tests/network/vmi_istio.go:152 Unexpected Warning event received: testvmi-2twdr,c3947566-91a2-415c-95f7-83cd5367ce16: server error. command SyncVMI failed: "LibvirtError(Code=1, Domain=0, Message='internal error: Child process (passt --one-off --socket /var/run/libvirt/qemu/run/passt/1-kubevirt-test-defaul-ua-default.socket --pid /var/run/libvirt/qemu/run/passt/1-kubevirt-test-defaul-ua-default-passt.pid --interface eth0 --log-file /var/run/kubevirt/passt.log --tcp-ports '~15000,~15001,~15004,~15006,~15008,~15009,~15020,~15021,~15053,~15090,1500,1501,22') unexpected exit status 1: Failed to bind port 22 (Permission denied) for option '-t ~15000,~15001,~15004,~15006,~15008,~15009,~15020,~15021,~15053,~15090,1500,1501,22', exiting\n')" Expected <string>: Warning not to equal <string>: Warning tests/watcher/watcher.go:195</failure>
```
Last time I ran these tests they passed, I never saw such failures before tbh.
@brianmcarey can you share the failing prow job link? 

I will try to reproduce it with the images you mentioned and update.

ormergi commented:
According to https://github.com/kubevirt/kubevirtci/pull/1331#issuecomment-2636048471 attached artifacts we see istio+passt tests are failing due to bug in Istio.
I posted a PR to quarantine the failing tests, once we have a workaround or bug fix we can un-quarentine them.
PTAL https://github.com/kubevirt/kubevirt/pull/13924

ormergi commented:
Hi, I managed to make istio 1.24.3 work with the following config:
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istio-operator
spec:
  profile: demo
  components:
    cni:
      enabled: true
      namespace: kube-system
      k8s:
        securityContext:
          seLinuxOptions:
            type: spc_t
  values:
    global:
      jwtPolicy: third-party-jwt
    cni:
      chained: false
      cniBinDir: /opt/cni/bin
      cniConfDir: /etc/cni/multus/net.d
      excludeNamespaces:
       - istio-system
       - kube-system
      logLevel: debug
    pilot:
      cni:
        enabled: true
        provider: "multus"
    sidecarInjectorWebhook:
      injectedAnnotations:
        k8s.v1.cni.cncf.io/networks: istio-cni
```
Missing part was setting istio CNI pods with seLinuxOptions spc_t.
@aerosouund @brianmcarey

aerosouund commented:
@ormergi 
Great work! Will push this to the branch ASAP. Do you prefer to stick with 1.23 or do we use 1.24? We already had a working fix for the privileged daemonset problem (although it wasn't ideal)

ormergi commented:
> . Do you prefer to stick with 1.23 or do we use 1.24?

I prefer newer version, but if its a trouble we can start with 1.23

aerosouund commented:
/restest-required

aerosouund commented:
/retest-required

brianmcarey commented:
/test check-provision-k8s-1.32

ormergi commented:
@aerosouund @brianmcarey hi could you please let use know what is the plan for Istio bump?

brianmcarey commented:
> @aerosouund @brianmcarey hi could you please let use know what is the plan for Istio bump?

If the sig-network tests in kubevirt are all happy - I think we should be good to take this in. Can you give it a once over review?

ormergi commented on cluster-provision/gocli/opts/istio/manifests/istio-operator-with-cnao.cr.yaml:
```diff
@@ -5,20 +5,29 @@ metadata:
   name: istio-operator
 spec:
   profile: demo
-  hub: quay.io/kubevirtci
```
Dont we need to work with kubevirt CI mirror, why was it dropped?

ormergi commented on cluster-provision/gocli/opts/istio/manifests/istio-operator-with-cnao.cr.yaml:
```diff
@@ -5,20 +5,29 @@ metadata:
+      cni:
+        enabled: true
+        provider: "multus"
+    sidecarInjectorWebhook:
```
Please add the following comment:
```suggestion
# override the injected annotation to mitigate the issue https://github.com/kubevirt/kubevirt/issues/13927
# Once the issue is resolved this configuration should be removed.
    sidecarInjectorWebhook:
```

ormergi commented on cluster-provision/gocli/opts/istio/manifests/istio-operator-with-cnao.cr.yaml:
```diff
@@ -5,20 +5,29 @@ metadata:
+        provider: "multus"
+    sidecarInjectorWebhook:
+      injectedAnnotations:
+        k8s.v1.cni.cncf.io/networks: istio-cni
```
Please add missing newline at the end of the file.

ormergi commented on cluster-provision/gocli/opts/istio/manifests/istio-operator.cr.yaml:
```diff
@@ -5,7 +5,6 @@ metadata:
 spec:
   profile: demo
-  hub: quay.io/kubevirtci
   components:
```
Please apply the same changes as in the `istio-operator-with-cnao.cr.yaml`, minus the `sidecarInjectorWebhook` part.

ormergi commented:
> If the sig-network tests in kubevirt are all happy - I think we should be good to take this in. Can you give it a once over review?

I dont know if this PR current state pass sig-network lanes, was it rested with this PR latest version?

I left some comments in regard to Istio manifests, PTAL.

brianmcarey commented:
> > If the sig-network tests in kubevirt are all happy - I think we should be good to take this in. Can you give it a once over review?
> 
> I dont know if this PR current state pass sig-network lanes, was it rested with this PR latest version?
> 
> I left some comments in regard to Istio manifests, PTAL.

Thanks @ormergi - once the comments are addressed - I will build a new provider and run the sig-network suite.

ormergi commented:
> Hey @aerosouund, why do we still want to install the old Istio Operator that is [deprecated](https://istio.io/latest/blog/2024/in-cluster-operator-deprecation-announcement/)? cc @ormergi

See this comment: 
```
This deprecation only affects users of the [In-Cluster Operator](https://archive.istio.io/v1.23/docs/setup/install/operator/). Users who install Istio with the istioctl install command and an IstioOperator YAML file are not affected.
```

ormergi commented:
In kubevirt CI we dont install istio opeator, the operator that install Istio https://istio.io/v1.23/docs/setup/install/operator/.
The method we use is istioctl + config file, the config file has come to be `IstioOperator` CR.
istioctl mange the installation, and the IstioOperator CR is not actually instantiated in the cluster.

oshoval commented:
about `KUBEVIRT_WITH_MULTUS_V3` 
dont we care more for V4 ?

ormergi commented:
We do, once CI transition to v4.

brianmcarey commented:
I build a new provider based on this PR and tested it against the kubevirt istio tests. The tests pass so we should try to introduce this. 

@aerosouund can you address Or's comments? 

I mirrored the images to kubevirtci quay.io repo so the images should be:
```
quay.io/kubevirtci/install-cni:1.24.1
quay.io/kubevirtci/pilot:1.24.1
quay.io/kubevirtci/proxyv2:1.24.1
```
This should be good to go following these small updates. 

Also I looked to see if the operator is deployed and I don't see it:
```
./kubevirtci/cluster-up/kubectl.sh get pods --all-namespaces | grep istio
WARNING: KUBEVIRTCI_GOCLI_CONTAINER is set and will take precedence over the also set KUBEVIRTCI_TAG
selecting podman as container runtime
istio-system             istio-egressgateway-84bfcbdd68-24j75                  1/1     Running   0             56m
istio-system             istio-ingressgateway-dd9f7f444-8jr67                  1/1     Running   0             56m
istio-system             istiod-b877844fb-md5m6                                1/1     Running   0             56m
kube-system              istio-cni-node-2rrnf                                  1/1     Running   0             56m
kube-system              istio-cni-node-ltnkk                                  1/1     Running   0             56m
kube-system              istio-cni-node-rrzrq                                  1/1     Running   0             56m
```

akalenyu commented:
> Don't we want to try and bump to 1.25 ?

If it's possible to do that in a later iteration, that would work better since https://github.com/kubevirt/kubevirt/pull/13422 is blocked over this

mhenriks commented:
> > Don't we want to try and bump to 1.25 ?
> 
> If it's possible to do that in a later iteration, that would work better since [kubevirt/kubevirt#13422](https://github.com/kubevirt/kubevirt/pull/13422) is blocked over this

I'm obviously with @akalenyu on this, would like to make movement on that pr

aerosouund commented:
@brianmcarey
Sure. i will address them and update you ASAP!

aerosouund commented:
@ormergi 

I have pushed the changes you suggested. need your help with taking one last look at `cluster-provision/gocli/opts/istio/manifests/istio-operator.cr.yaml` as i am not sure what changes should apply to it and what doesn't. If its OK please leave a LGTM

cc: @brianmcarey

ormergi commented:
It seem `opts/istio/manifests/istio-operator.cr.yaml` was never meant to work with Multus, and in practice its not in use because we CI always deploy CNAO on sig-network lane.
Makes me wonder why do we even need this manifest in the first place.Anyhow we can address that later on a follow-up PR.

Thanks for the changes.
LGTM on Istio manifests part
/lgtm

ormergi commented:
@nirdothan FWIW there is no available release of Istio including the bugfix for  https://github.com/kubevirt/kubevirt/issues/13927, not event v1.25. Once there is, we can bump Istio.

brianmcarey reviewed (commented):
Thanks @aerosouund - looks good. I just have one small comment on provision.sh.

aerosouund commented:
@brianmcarey 
Removed. but kubevirtCI's CI will fail because the `shared_vars.sh` script in the nodes is still the old version. the provision phase is when it gets created and thats why i had removed it to make CI pass and was going to submit a patch afterwards to revert this

brianmcarey commented:
> @brianmcarey Removed. but kubevirtCI's CI will fail because the `shared_vars.sh` script in the nodes is still the old version. the provision phase is when it gets created and thats why i had removed it to make CI pass and was going to submit a patch afterwards to revert this

Ok makes sense - lets drop the last commit and we can handle that in a followup

brianmcarey reviewed (approved):
/approve 

thanks for all the work on this @aerosouund !

brianmcarey commented on cluster-provision/k8s/1.31/provision.sh:
```diff
@@ -52,7 +52,8 @@ export PATH="$ISTIO_BIN_DIR:$PATH"
   mkdir -p "$ISTIO_BIN_DIR"
-  curl "https://storage.googleapis.com/kubevirtci-istioctl-mirror/istio-${ISTIO_VERSION}/bin/istioctl" -o "$ISTIO_BIN_DIR/istioctl"
+  curl -L  https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-linux-amd64.tar.gz -O
+  tar -xvf ./istio-${ISTIO_VERSION}-linux-amd64.tar.gz --strip-components=2 -C ${ISTIO_BIN_DIR} istio-${ISTIO_VERSION}/bin/istioctl
```
This needs to be applied to the 1.32 provider & 1.33 provider 

/lgtm cancel

aerosouund replied:
Will do

brianmcarey reviewed (commented):
/lgtm

ormergi commented:
@aerosouund ^

brianmcarey commented:
> Placing hold to give chance for resolving this comment https://github.com/kubevirt/kubevirtci/pull/1331/files#r2006316352
> 
> Feel free to remove when its done /hold

Thanks @ormergi  for double checking - we are going to take that in a followup - https://github.com/kubevirt/kubevirtci/pull/1331#issuecomment-2745883770

/hold cancel
